internal/services/containerapps/container_app_environment_resource.go

```diff
@@ -90,11 +91,18 @@ func (r ContainerAppEnvironmentResource) Arguments() map[string]*pluginsdk.Schem
 		"log_analytics_workspace_id": {
 			Type:         pluginsdk.TypeString,
 			Optional:     true,
-			ForceNew:     true,
 			ValidateFunc: workspaces.ValidateWorkspaceID,
 			Description:  "The ID for the Log Analytics Workspace to link this Container Apps Managed Environment to.",
 		},
 
+		"logs_destination": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringInSlice([]string{"log-analytics", "azure-monitor"}, false),
+			Description:  "The destination for the application logs. Possible values are `log-analytics` or `azure-monitor`.",
+		},
```

Comment:
Two things here:
1. Rather than making this Computed, which'd mask any default value for this field - can we default this here until 4.0 (at which point we can make this field required).
2. Whilst I can see that the `Description` field is a string value, do you have a canonical source for the casing of the values `log-analytics` and `azure-monitor` - is that what the API is returning/defined somewhere? Typically we source these from `hashicorp/go-azure-sdk`, but since this is a String field we don't have a backing constant for that, so it'd be good to confirm the casing (and file an upstream issue to fix this, once we have that available)

```suggestion
		// TODO: make this Required in 4.0
		"logs_destination": {
			Type:         pluginsdk.TypeString,
			Optional:     true,
			ValidateFunc: validation.StringInSlice([]string{"log-analytics", "azure-monitor"}, false),
			Description:  "The destination for the application logs. Possible values are `log-analytics` or `azure-monitor`.",
		},
```

Reply:
Alternatively we could infer the value for this field based on `log_analytics_workspace_id` having a value, which is probably better here, since there's only 2 sources?

Reply:
The reference to the strings comes from the [REST API documentation](https://learn.microsoft.com/en-us/rest/api/containerapps/managed-environments/create-or-update?view=rest-containerapps-2024-03-01&tabs=HTTP#applogsconfiguration), though it is worth noting that the documentation implies that `none` is a valid value, when the actually mean that the object is empty. This is more clear in the API error message which doesn't quote none.